Allow individual pixel brightness to be set.

jmb opened the pull request:
This change adds an optional parameter to the setPixel and setPixelRGB functions which allows individual pixel brightness settings to made. If left off, the default is to use the globalBrightness value.

jmb committed:
Allow individual pixel brightness to be set.

This change adds an optional parameter to the setPixel and setPixelRGB
functions which allows individual pixel brightness settings to made. If
left off, the default is to use the globalBrightness value.

tinue commented:
Thank you for the pull request! Right now, I have no Pi Zero soldered to the strips, so I can't test your changes. They look good, but I would still like to see it working first ;) So please give me some time, before I merge.

One more thing: The global brightness is very useful to limit the current with large strips. So an even better change would be to have the local brightness relative to the global one. For example, if the local brightness is 31, the pixel would have the global brightness value (say 10). If it is 15, then the value would be 5. For full control, one would simply set the global brightness to 31.

jmb commented:
Great idea, I'll have a look at my code and update - should be pretty straightforward.

jmb committed:
Limit the pixel brightness to be relative to the global brightness setting.

The global brightness is very useful to limit the current with large
strips, so now the individual pixel brightness is set relative to the
global brightness (rounding up to the nearest integer on the basis that
some light is expected unless the brightness is set to 0).

Also some basic bounds checking (and adjustment) is included for the
global brightness.
